apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.SingleChannel.cs

```diff
@@ -407,8 +435,40 @@ private void HandlePullMessageData(Task<bool> moveNextTask)
                 _pullBackoff = TimeSpan.FromSeconds(0.5);
                 StartStreamingPull();
             }
-        }
 
+            // This local method is used exclusively in exactly-once delivery flow.
+            // The method processes and shares the messages that had successful receipt ModAcks.
+            // Additionaly, it removes Ack IDs of messages that failed to extend successfully from the given list of messages.
+            Task ProcessSuccessfulMessages(List<ReceivedMessage> messages, HashSet<string> messageIds)
+            {
+                // Remove all Ack IDs that failed to extend successfully.
+                // The value of false in the dictionary represents the failure.
+                var failures = _receiptModAckStatusLookup.Where(keyValue => keyValue.Value == false).Select(keyValue => keyValue.Key).ToList();
+                if (failures.Any())
+                {
+                    // Remove failed messages.
+                    var ackIdsToRemove = new HashSet<string>(failures);
+                    _ = messageIds.RemoveWhere(ackIdsToRemove.Contains);
```

Comment:
I was just trying to avoid new editorconfig messages, specifically "Expression Value is never used." I have now removed the discard and not assigning the result to anything.